sentry/src/main/java/io/sentry/TraceContext.java

```diff
@@ -50,8 +59,39 @@ public final class TraceContext implements JsonUnknown, JsonSerializable {
         new Dsn(sentryOptions.getDsn()).getPublicKey(),
         sentryOptions.getRelease(),
         sentryOptions.getEnvironment(),
-        user != null ? new TraceContextUser(user) : null,
-        transaction.getName());
+        user != null ? user.getId() : null,
+        user != null ? getSegment(user) : null,
+        transaction.getName(),
+        sampleRateToString(sentryOptions.getTracesSampleRate()));
+  }
+
+  private static @Nullable String getSegment(final @NotNull User user) {
+    final Map<String, String> others = user.getOthers();
+    if (others != null) {
+      return others.get("segment");
+    } else {
+      return null;
+    }
+  }
+
+  private static @Nullable String sampleRateToString(@Nullable Double sampleRateAsDouble) {
+    if (sampleRateAsDouble == null) {
+      return null;
+    }
+
+    if (sampleRateAsDouble == null
+        || sampleRateAsDouble.isNaN()
+        || sampleRateAsDouble.isInfinite()) {
+      return null;
+    }
+
+    if (sampleRateAsDouble < 0.0 || sampleRateAsDouble > 1.0) {
+      return null;
+    }
+
+    DecimalFormat df =
+        new DecimalFormat("#.################", DecimalFormatSymbols.getInstance(Locale.US));
+    return df.format(sampleRateAsDouble);
```

Comment:
We should avoid `US` locale and use `Locale.ROOT` instead, some custom OS versions might not have the US locale.

Reply:
Because we don't want exponent formatting. `0.00000021.toString()` would become `2.1E-7` which we don't want.
Can replace the locale.

Reply:
Is the spec defining the number of max digits? because `################` can lead to a different value, not sure if this would round or not.

Reply:
It would round on the last digit. Yes we're aligning with JS on the max digits here.

Reply:
Updated the locale usage